### PR TITLE
DEP: Deepdish to h5py

### PR DESCRIPTION
**Pull request by carleyjmartin**
# Scope 

The deepdish library is not being maintained any longer so we are going to move to using h5py solely. Instances of deeepdish use in the borealis package need updating. This PR is to track those changes.

**issue:** #58 

## Approval

**Number of approvals:** 2

## Test

Will provide when testing needed. 

*Reminder:
  - Modification line to be filled in *

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -49,8 +49,8 @@
     author="SuperDARN",
     include_package_data=True,
     setup_requires=['pyyaml', 'numpy',
-                    'h5py', 'deepdish', 'pathlib2'],
+                    'h5py>=3.3.0', 'deepdish', 'pathlib2'],
```
Comment (Lozzy-Bear) [inline setup.py:52]:
```suggestion
                    'h5py>=3.3.0', 'pathlib2'],
```

File: setup.py [hunk 2]
```diff
@@ -49,8 +49,8 @@
...
     # pyyaml library install
     install_requires=['pyyaml', 'numpy',
-                      'h5py', 'deepdish', 'pathlib2']
+                      'h5py>=3.3.0', 'deepdish', 'pathlib2']
```
Comment (Lozzy-Bear) [inline setup.py:55]:
```suggestion
                      'h5py>=3.3.0', 'pathlib2']
```

File: pydarnio/borealis/borealis_array.py [hunk 3]
```diff
@@ -38,7 +38,9 @@
 files, see: https://borealis.readthedocs.io/en/latest/
 """
 import deepdish as dd
```
Comment (tjk584) [inline pydarnio/borealis/borealis_array.py:40]:
```suggestion
```

File: pydarnio/borealis/base_format.py [hunk 4]
```diff
@@ -1102,15 +1102,15 @@ class methods used inside this method should be specific
...
+            if datatype == str:
                 # unicode type needs to be explicitly set to have
                 # multiple chars (256)
                 datatype='|U256'
             empty_array = np.empty(array_dims, dtype=datatype)
             # Some indices may not be filled due to dimensions that are maximum values (num_sequences, etc. can change
             # between records), so they are initialized with a known value first.
             # Initialize floating-point values to NaN, and integer values to -1.
-            if datatype is np.int64 or datatype is np.uint32:
+            if datatype is np.int64 or datatype is np.uint32 or datatype is np.uint8:
```
[the remark this answers is not in the document]
Comment (Lozzy-Bear) [reply pydarnio/borealis/base_format.py:1113]:
```suggestion
            if datatype == np.int64 or datatype == np.uint32 or datatype == np.uint8:
```

[the remark this answers is not in the document]
Comment (Lozzy-Bear) [reply pydarnio/borealis/base_format.py:1113]: '==' compares variables, whereas 'is' checks if they point to the same location in memory. I think we need '==' technically, but the result is the same.

File: pydarnio/borealis/borealis_restructure.py [hunk 5]
```diff
@@ -374,7 +378,7 @@ def _site_to_array_restructure(self):
                                 # Initialize array now with correct data type.
                                 dtype = self.format.single_element_types()[field]
                                 new_data_dict[field] = np.empty(num_records, dtype=dtype)
-                                if dtype is np.int64 or dtype is np.uint32:
+                                if dtype is np.int64 or dtype is np.uint32 or dtype is np.uint8:
```
Comment (Lozzy-Bear) [inline pydarnio/borealis/borealis_restructure.py:381]:
```suggestion
                                if dtype in [np.int64, np.uint32, np.uint8]:
```

**Review by Lozzy-Bear — COMMENTED** [review]
Looks good. I left some code change comments that shorten up the if statements.

File: pydarnio/borealis/borealis_array.py [hunk 6]
```diff
@@ -37,8 +37,9 @@
 For more information on Borealis data files and how they convert to SDarn
 files, see: https://borealis.readthedocs.io/en/latest/
 """
-import deepdish as dd
+import h5py
 import logging
+import numpy as np
```
Comment (carleyjmartin) [inline pydarnio/borealis/borealis_array.py:42]: Numpy not used in file?

**Review by carleyjmartin — COMMENTED** [review]
No issues with 0.6.1 or 0.5 files given, and I've just thrown all borealis code I have at it with no errors. (I can't tick approve on my own PRs, but I approve 💯)

**Pull request merged**